### DISCUSSION TITLE
Assay missing after subset from scATAC object by feature

### DISCUSSION THREAD

**Discussion by RitataLU** [stuart-lab/signac#520] [Q&A] [answered]
Hi Signac team,

I would like to perform the same idea from scRNA clustering with features (select_genes) on scATAC-seq, like in scRNA, there is a step after subset object,
`object <- RunPCA(object = object, features=selected_genes, npcs = 30, verbose = FALSE)`, and then do the `RunUMAP`,`FindNeighbors`,`FindClusters`.

I would like to use the same way on scRNA, clustering without some genes, but I couldn't find the similar way on RunSVD or RunUmap.., so what I did is `sub_atac_object <- subset (atac_object, feature = selected_genes)`
and then do the standard method on clustering..,
but I found that after I got the new clusters in sub_atacobject, I couldn't do the find motif between clusters. since it looks like some assay missing. 
originally ataco_bject has four assays, RNA, peaks,peaksinters, chromvar, but sub_atac_object only has RNA assay.

I was wondering how could I fix this issue, or do you have suggestions on my processing steps?

Thanks!
Rita

**Comment by timoast** [comment, marked as answer]
Those other assays are probably removed because they don't contain any of the features you asked to keep.

If you want to run SVD using a pre-defined set of features, you can use the `features` argument in `RunSVD()`. See the documentation for `RunSVD()` for more information.
